### PR TITLE
Add AzurePipelinesCredential parameters to its constructor

### PR DESCRIPTION
**Pull request by chlowell**
This is the planned final constructor API. I'll update markdown docs in another PR.

### CODE REVIEW

File: sdk/azidentity/azure_pipelines_credential.go [hunk 1]
```diff
@@ -48,28 +46,36 @@ type AzurePipelinesCredentialOptions struct {
...
 	if u == "" {
 		return nil, fmt.Errorf("no value for environment variable %s. This should be set by Azure Pipelines", systemOIDCRequestURI)
 	}
-	sat := os.Getenv(systemAccessToken)
-	if sat == "" {
-		return nil, errors.New("no value for environment variable " + systemAccessToken)
+	if clientID == "" {
+		return nil, fmt.Errorf("no client ID specified")
+	}
+	if serviceConnectionID == "" {
```
Comment (jhendrixMSFT) [inline sdk/azidentity/azure_pipelines_credential.go:69]: Does `systemAccessToken` need similar validation?

**Review by jhendrixMSFT — APPROVED** [review]
I assume the changelog will be updated in the separate docs PR.

**Comment by chlowell** [comment]
Yes, I'll update the changelog after publishing v1.6.0 (this PR will ship in v1.7.0-beta.1)

File: sdk/azidentity/azure_pipelines_credential.go (hunk 1 above)
Comment (chlowell) [reply sdk/azidentity/azure_pipelines_credential.go:69]: 🤦 yep, that would be for the best

**Pull request merged**